packages/runtime/src/next/api-handler/config/app-router.ts

```diff
@@ -14,15 +14,18 @@ import { MakeswiftClient } from '../../../client'
 type Params = { [key: string]: string | string[] }
 type Context = { params: Promise<Params> }
 
-export type ApiHandlerArgs = [NextRequest, Context]
-export const argsPattern = [P.instanceOf(NextRequest), P.any] as const
+export type ApiHandlerArgs = [NextRequest | Request, Context]
```

Comment:
I think we should add a comment here, something along the lines of

> In older versions of Next (prior to 15.5.0), the inbound request is typed as a `NextRequest`, but requests received at runtime fail the `instanceof NextRequest` check. In newer versions, runtime requests are correctly typed as `NextRequest`. To maintain compatibility across all Next.js versions, we match against both.

Reply:
Although, even though it's marked as a fix, one can argue that this change contradicts my statement about types: https://github.com/vercel/next.js/pull/82644/files.

Nevertheless, the requests received at runtime must have been at least duck-typed as `NextRequest`; otherwise, accessing `request.nextUrl` would not have been possible.